### DISCUSSION TITLE
Best method to debug using VS2022?

### DISCUSSION THREAD

**Discussion by aoqia194** [YimMenu/YimMenu#617] [Q&A] [answered]
I am writing some code for YimMenu and I am getting an access violation (I am presuming it's probably an indexing issue with arrays I have) but I can't really get much detailed information from the stack and VS debugger gets detected by the game, so that's out of the way.

I am trying to debug in Story Mode, but I am not specifically choosing SP (only because it's easier)
*MY QUESTION*: How would I successfully go about debugging GTA so that it can provide useful information to me? For example, catching unhandled exceptions?

EDIT: I have already tried x64dbg w/ ScyllaHide and although it works, I would rather use VS2022's built in debugger if I could

**Comment by ghost** [comment]
Don't make a bunch of changes. make one change, build, fail, fix, build and move on. I recommend making use of LOG'ing if you have a guess where the issue will be or just use what g3log gives you. Google and Stackoverflow will always be your friend.

**Reply by aoqia194** [reply]
I've tried logging basically everything (Native return values, for loop indexes, array indexing, etc) and they all return normal

**Reply by ghost** [reply]
> I've tried logging basically everything (Native return values, for loop indexes, array indexing, etc) and they all return normal
try doing what the others said. or 

if you want to be a monkey. LOG each line until the game crashes 🤓

**Reply by aoqia194** [reply]
> ~snip~

Don't worry, I've just about tried that too ;-;

**Reply by aoqia194** [reply]
Also how tf you reply that fast u speedy person

**Reply by ghost** [reply]
> Also how tf you reply that fast u speedy person

i have github installed on my phone insta-notifs + im just chillin in bed

**Comment by xiaoxiao921** [comment]
you can attach vs i'm pretty sure, you just need to patch their check (very much like scillahide is doing) I don't have them at hand but just know that its possible

also fyi if you are debugging your menu in SP you can also just use symbols + pdb

**Comment by FIying-Scotsman** [comment, marked as answer]
You can attach VS's debugger, yes. You've just got to go through a few hoops first to get it working.

Using x64Dbg, attach scyllahide to GTA and press "continue" a few times to get past the game freezing as it throws a couple of exceptions (which triggers the debugger's "pause" feature).

Once the game stops freezing, _detach_ x64Dbg (although best to leave open in case the game crashes and you've got to re-attach it again) as ScyllaHide's DLL has been injected into GTA and is currently hiding any debuggers you may want to attach. 

You can now attach VS's debugger and start setting breakpoints etc. 

You can also look at hooking the "is debugger present" WinAPI and/or patching socialclub.dll's debugger checks as that's where the debugger related code is (unless they've moved it).

**Reply by aoqia194** [reply]
I honestly thought YimMenu would have already patched the debugger checks but it makes sense as YimMenu isn't hooking any debugger of sorts.

**Reply by aoqia194** [reply]
Hey I got it working with this! So glad I can see the local variables and their types (Also don't forget to use the RelWithDebInfo and not the Release dll so you don't go crazy like I did)💕
